internal/server/ranking.go

```diff
@@ -112,6 +143,31 @@ func (a byRank) Less(i, j int) bool {
 	if scorei != scorej {
 		return scorei < scorej
 	}
+
+	datesi := []string{}
+	for date := range a[i].Val {
+		datesi = append(datesi, date)
+	}
+	sort.Strings(datesi)
+	latesti := datesi[len(datesi)-1]
+
+	datesj := []string{}
+	for date := range a[j].Val {
+		datesj = append(datesj, date)
+	}
+	sort.Strings(datesj)
+	latestj := datesj[len(datesj)-1]
+
+	// Series with latest data is ranked higher
+	if latesti != latestj {
+		return latesti > latestj
+	}
+
+	// Series with more data is ranked higher
+	if len(datesi) != len(datesj) {
+		return len(datesi) > len(datesj)
+	}
```

Comment:
if the maps here and above are of the same, maybe we can use a helper?

Reply:
The logic is repeated since one uses Go struct and the other using protobuf. I have a pending task to use protobuf for all the API and make them /v1/..